activation/atxdb.go

```diff
@@ -82,46 +174,16 @@ func (db *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, er
 		return count, nil
 	}
 
-	var counter uint32 = 0
-	set := make(map[types.AtxId]struct{})
 	pubEpoch := a.PubLayerIdx.GetEpoch(db.LayersPerEpoch)
 	if pubEpoch < 1 {
 		return 0, fmt.Errorf("publication epoch cannot be less than 1, found %v", pubEpoch)
 	}
 	countingEpoch := pubEpoch - 1
 	firstLayerOfLastEpoch := types.LayerID(countingEpoch) * types.LayerID(db.LayersPerEpoch)
 
-	traversalFunc := func(blkh *types.BlockHeader) error {
-		blk, err := db.meshDb.GetBlock(blkh.Id)
-		if err != nil {
-			db.log.With().Error("cannot validate atx, block wasn't found", log.AtxId(a.ShortId()), log.BlockId(uint64(blkh.Id)))
-			return err
-		}
-		//skip blocks not from atx epoch
-		if blk.LayerIndex.GetEpoch(db.LayersPerEpoch) != countingEpoch {
```

Comment:
you've omitted this check in your code, which has a performance cost, but also means you're counting ATXs that may not have been referenced in the epoch you're supposed to be counting them in.

Reply:
Fixed